gwy/templates/go.mod.tmpl

```diff
@@ -9,6 +9,6 @@ require (
 	github.com/sirupsen/logrus v1.7.0
 	github.com/spf13/pflag v1.0.5
 	github.com/spf13/viper v1.7.1
-	google.golang.org/grpc v1.34.0
-	google.golang.org/protobuf v1.24.0
+	google.golang.org/grpc v1.40.0
```

Comment:
Why `v1.40.0` not `v1.42.0`?

Reply:
I think this is where @ido-namely asked that I back this down from 1.42.0 so as to allow other languages (whose gRPC generators are not up to date yet) to work.

Does that sound right?

Reply:
not exactly, I think we need our versions to be aligned, if this is a 1.42 image we're releasing, then this package should be 1.42 as well.
If anyone would like to use 1.40, they should use the 1.40 image.
1.42 will be the first Go 1.17 based image.

Reply:
Updated PR accordingly (after merging your upgrade to 1.41.0 and resolving conflicts).